### PR TITLE
Implement XHR assertSuccess skipping

### PR DESCRIPTION
**Pull request by jridgewell**
Supercedes https://github.com/ampproject/amphtml/pull/9448.
Fixes https://github.com/ampproject/amphtml/issues/8017.

/cc @taymonbeal

### CODE REVIEW

**Comment by cramforce** [comment]
This change is not versionskew safe, but probably fine since it is exceptional.

**Comment by jridgewell** [comment]
I can split it into two PRs pretty easily, and just wait a week on the second PR.

File: extensions/amp-form/0.1/amp-form.js [hunk 1]
```diff
@@ -488,13 +493,14 @@ export class AmpForm {
...
-   * @param {!Error} error
+   * @param {?../../../src/service/xhr-impl.FetchResponse} response
    * @private
    */
-  handleXhrSubmitFailure_(error) {
+  handleXhrSubmitFailure_(response) {
     let promise;
-    if (error && error.response) {
-      promise = error.response.json().catch(() => null);
+    // TODO
```
Comment (cvializ) [inline extensions/amp-form/0.1/amp-form.js]: hm?

File: extensions/amp-form/0.1/amp-form.js [hunk 2]
```diff
@@ -504,8 +510,8 @@ export class AmpForm {
       this.cleanupRenderedTemplate_();
       this.setState_(FormState_.SUBMIT_ERROR);
       this.renderTemplate_(responseJson || {});
-      this.maybeHandleRedirect_(error.response);
-      user().error(TAG, `Form submission failed: ${error}`);
+      this.maybeHandleRedirect_(response);
+      user().error(TAG, `Form submission failed`);
```
Comment (cvializ) [inline extensions/amp-form/0.1/amp-form.js]: Is there any way to print the error here, or something helpful to describe how the request failed?

**Comment by taymonbeal** [comment]
Has this PR been abandoned?

**Comment by cvializ** [comment]
Closing as it appears abandoned.

**Pull request closed without merging**